taichi/rhi/metal/metal_rhi_enum_mappings.h

```diff
@@ -0,0 +1,121 @@
+#ifdef VERTEX_FORMATS
```

Comment:
I would just make a header and a cpp file, where in the headers it contains:

```c++
const BufferFormat mtl_vertex_format_to_buffer_format();
...
```

And then implement them in a cpp file. This way we don't generate a whole bunch of identical table for every function that uses it